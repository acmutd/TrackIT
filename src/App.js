import React from "react";
import Admin from "./Components/Admin/Admin";
import User from "./Components/User/User";
import Pricing from "./Components/Pages/Pricing";
import Error404 from "./Components/Pages/Error404";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import firebase from "firebase";

const firebaseConfig = {
  apiKey: "AIzaSyC10N5kRDieKncmUESxswqkYQ_359f9Qes",
  authDomain: "trackit-271619.firebaseapp.com",
  databaseURL: "https://trackit-271619.firebaseio.com",
  projectId: "trackit-271619",
  storageBucket: "trackit-271619.appspot.com",
  messagingSenderId: "972365141905",
  appId: "1:972365141905:web:fbda064275f635298cec30",
  measurementId: "G-HRLPFBGB1E",
};

let app = firebase.initializeApp(firebaseConfig);
let db = app.firestore();

function App() {
  return (
    <div>
      {/* replace this with <User /> when wanting to test out the user side, we can decide at a later point when the admin side should appear and when the user side should apper */}
      <Switch>
        <Route path="/" render={(props) => <User database={db} />} exact />
        <Route
          path="/admin"
          render={(props) => <Admin database={db} />}
          exact
        />
        <Route path="/pricing" component={Pricing} exact />
        <Route path="*" component={Error404} exact />
      </Switch>
    </div>
  );
}

export default App;
