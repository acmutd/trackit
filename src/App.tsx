import * as React from "react";
import Admin from "./Components/Admin/Admin";
import User from "./Components/User/User";
import Pricing from "./Components/Pages/Pricing";
import Error404 from "./Components/Pages/Error404";
import { Route, Switch, BrowserRouter } from "react-router-dom";

function App() {
  return (
    <div>
      <BrowserRouter>
        <Switch>
          <Route
            path="/"
            render={(props) => <User />}
            exact
          />
          <Route
            path="/admin"
            render={(props) => <Admin />}
            exact
          />
          <Route path="/pricing" component={Pricing} exact />
          <Route path="*" component={Error404} exact />
        </Switch>
      </BrowserRouter>
    </div>
  );
}

export default App;
