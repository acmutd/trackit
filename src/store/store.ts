import { createStore } from "redux";
import appState from "../reducers/rootReducer";

const store = createStore(appState);
export default store;
