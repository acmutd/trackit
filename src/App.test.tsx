import * as React from "react";
import { shallow, mount } from "enzyme";
import App from "./App";
import { Provider } from "react-redux";
import store from "./store/store";

// describe("Render testing", () => {
//   it("shallow render without crashing", () => {
//     const component = shallow(
//       <Provider store={store}>
//         <App />
//       </Provider>
//     );

//     expect(component).toMatchSnapshot();

//     component.unmount();
//   });

//   it("full render to DOM without crashing", () => {
//     const component = mount(
//       <Provider store={store}>
//         <App />
//       </Provider>
//     );

//     expect(component).toMatchSnapshot();

//     component.unmount();
//   });
// });

// unit tests for env file
describe("env", () => {
  it("should have an api key", () => {
    expect(process.env.REACT_APP_DEV_API_KEY).toBeDefined();
  });
  it("should have an auth domain", () => {
    expect(process.env.REACT_APP_DEV_AUTH_DOMAIN).toBeDefined();
  });
  it("should have a database url", () => {
    expect(process.env.REACT_APP_DEV_DATABASE_URL).toBeDefined();
  });
  it("should have a project id", () => {
    expect(process.env.REACT_APP_DEV_PROJECT_ID).toBeDefined();
  });
  it("should have a storage bucket", () => {
    expect(process.env.REACT_APP_DEV_STORAGE_BUCKET).toBeDefined();
  });
  it("should have a messenger sender id", () => {
    expect(process.env.REACT_APP_DEV_MESSENGER_SENDER_ID).toBeDefined();
  });
  it("should have a app id", () => {
    expect(process.env.REACT_APP_DEV_APP_ID).toBeDefined();
  });
  it("should have a measurement id", () => {
    expect(process.env.REACT_APP_DEV_MEASUREMENT_ID).toBeDefined();
  });

  it("should have an auth0 domain", () => {
    expect(process.env.REACT_APP_AUTH0_DOMAIN).toBeDefined();
  });
  it("should have an auth0 client id", () => {
    expect(process.env.REACT_APP_AUTH0_CLIENTID).toBeDefined();
  });
  it("should have an auth0 audience", () => {
    expect(process.env.REACT_APP_AUTH0_AUDIENCE).toBeDefined();
  });
});
