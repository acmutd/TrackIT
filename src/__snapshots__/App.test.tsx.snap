// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render testing full render to DOM without crashing 1`] = `ReactWrapper {}`;

exports[`Render testing shallow render without crashing 1`] = `ShallowWrapper {}`;
